conans/client/generators/__init__.py

```diff
@@ -186,7 +186,13 @@ def bat_content(files):
                  bat_content(deactivates(bats)))
         if ps1s:
             def ps1_content(files):
-                return "\r\n".join(['& "{}"'.format(b) for b in files])
+                content = "\r\n"
+                for b in files:
+                    if "vcvars" in b:
+                        content += 'if (-not $env:VSCMD_ARG_HOST_ARCH){{ & "{}" }}\n'.format(b)
```

Comment:
I would add this conditional in the contents of `conanvcvars.ps1` and keep the logic here as it already was? it means the file would "always" be called when activating `conanbuild.ps1`, but it would immediately skip the logic if the variable is defined.

As a second question: is this `VSCMD_ARG_HOST_ARCH` guaranteed to exist for all versions of visual studio supported by Conan 2.0?

Reply:
To the first question: this is something I thought about, I thought it would be better to not even call the other script, but it's true it is a good idea to remove the logic from here and leave it completely at the file generation.

Reply:
I agree with @jcar87 better keep this logic as-is, and move it to the other file, calling it is cheap, and it will be cleaner